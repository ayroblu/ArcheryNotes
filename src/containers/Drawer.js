// @flow
import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Platform,
} from 'react-native'
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'

import * as mainActions from '../actions/main'
import {textStyle} from '../config'
//import * as theme from '../theme'

class Drawer extends Component {
  _goTo(screen){
    // apparently normal navigator ONLY works in android
    // https://github.com/wix/react-native-navigation/issues/1143
    const {navigator} = Platform.OS==='ios' ? this.props.main : this.props
    navigator.push({screen: `arn.${screen}`, animationType: 'slide-horizontal'})
    navigator.toggleDrawer({side: 'left'})
  }
  _resetTo(screen){
    const {navigator} = Platform.OS==='ios' ? this.props.main : this.props
    navigator.resetTo({screen: `arn.${screen}`})
    navigator.toggleDrawer({side: 'left'})
  }
  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.line} onPress={()=>this._goTo('About')}>
          <Text style={styles.lineText}>About</Text>
        </TouchableOpacity>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    paddingTop: Platform.select({ios: 20, android: 0})
  },
  line: {
    marginLeft: 20
  , marginRight: 20
  , paddingLeft: 40
  , padding: 20
  , borderBottomWidth: 1
  , borderColor: '#ddd'
  , flexDirection: 'row'
  , alignItems: 'center'
  },
  lineText: {
    ...textStyle
  , fontSize: 20
  },
})

export default connect(state=>({
  main: state.main
}), dispatch=>({
  mainActions: bindActionCreators(mainActions, dispatch)
}))(Drawer)
