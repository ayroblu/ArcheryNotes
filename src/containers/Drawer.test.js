import 'react-native'
import React from 'react'
import {Provider} from 'react-redux'
import Drawer from './Drawer'
import configureStore from '../store'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const store = configureStore()

it('renders correctly', ()=>{
  const props = {
    navigator: {
      push(){}
    , toggleDrawer(){}
    , resetTo(){}
    }
  }
  const tree = renderer.create(
    <Provider store={store}>
      <Drawer {...props} />
    </Provider>
  )
  expect(tree).toMatchSnapshot()
})
