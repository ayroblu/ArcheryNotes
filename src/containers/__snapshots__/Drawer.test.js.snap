// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "marginLeft": 20,
        "marginRight": 20,
        "opacity": 1,
        "padding": 20,
        "paddingLeft": 40,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#555",
          "fontSize": 20,
          "fontWeight": "300",
        }
      }
    >
      About
    </Text>
  </View>
</View>
`;
